client/src/components/Tea/TeaEditor/TeaEditorContainer.jsx

```diff
@@ -173,15 +174,11 @@ export class TeaEditorContainer extends React.Component {
 }
 
 const mapStateToProps = (state, ownProps) => {
-  let currentTeaId = ownProps.match.params.id;
-  let currentTea = currentTeaId
-    ? find(state.teas, tea => tea.id === currentTeaId) || {}
-    : {};
   return {
     teas: state.teas,
     teaTypes: state.teaTypes,
     userID: state.auth.user.id,
-    currentTea: currentTea
+    currentTea: state.teas.allTeas[ownProps.match.params.id] || {}
```

Comment:
I don't recommend creating new objects or arrays within `mapStateToProps`. It leads to unnecessary re-renders because shallow comparison fails (i.e. `{} !== {}`). I'd recommend defining an `emptyObj` variable outside of this function and referring to it here.

Reply:
Good point, I hadn't thought of `{} !== {}` setting off re-renders any time there is no current tea.